railties/test/generators/argv_scrubber_test.rb

```diff
@@ -56,6 +57,21 @@ def self.default_rc_file
         assert_equal [], args
       end
 
+      def test_xdg_config_no_custom_rc
+        xdg = Dir.tmpdir
+        ENV["XDG_CONFIG_HOME"] = xdg
+        rails_xdg_path = File.join(xdg, "rails")
+        Dir.mkdir(rails_xdg_path, 0700) unless Dir.exist?(rails_xdg_path)
+        Dir.new(rails_xdg_path)
+        railrc_path = File.join(rails_xdg_path, "railsrc")
+        railsrc = File.new(railrc_path, "w")
+        railsrc.close
+
+        scrubber = Class.new(ARGVScrubber)
+        assert_equal scrubber.default_rc_file, railrc_path
+        File.delete(railrc_path) if File.exist?(railrc_path)
```

Comment:
@quintrino I hope you don't mind me piling on! :pray:  I felt like this test code could be a bit more focused:

```suggestion
      def test_default_rc_file_with_xdg_config_home
        Dir.mktmpdir do |dir|
          rc_file = File.join(dir, "rails/railsrc")
          FileUtils.mkdir_p(File.dirname(rc_file))
          FileUtils.touch(rc_file)

          switch_env("XDG_CONFIG_HOME", dir) do
            assert_equal rc_file, ARGVScrubber.default_rc_file
          end
        end
```

That will take care of cleaning up the file and the environment variable.  Though to use `switch_env` you'll need to add `require "env_helpers"` to the top of the file, and `include EnvHelpers` to the top of the test class ([similar to this](https://github.com/rails/rails/blob/1fd17336544a04119316c0a9dfc2b1464b4b3f30/railties/test/generators/actions_test.rb#L5-L9)).

Reply:
@jonathanhefner I've applied these changes now, let me know if you think it needs anything else.